core/datastore/src/main/java/com/google/samples/apps/nowinandroid/core/datastore/NiaPreferencesDataSource.kt

```diff
@@ -33,54 +31,89 @@ class NiaPreferencesDataSource @Inject constructor(
     val userDataStream = userPreferences.data
         .map {
             UserData(
-                bookmarkedNewsResources = it.bookmarkedNewsResourceIdsList.toSet(),
-                followedTopics = it.followedTopicIdsList.toSet(),
-                followedAuthors = it.followedAuthorIdsList.toSet(),
+                bookmarkedNewsResources = it.bookmarkedNewsResourceIdsMap.keys,
+                followedTopics = it.followedTopicIdsMap.keys,
+                followedAuthors = it.followedAuthorIdsMap.keys,
             )
         }
 
-    suspend fun setFollowedTopicIds(followedTopicIds: Set<String>) =
-        userPreferences.setList(
-            listGetter = { it.followedTopicIds },
-            listModifier = { followedTopicIds.toList() },
-            clear = { it.clear() },
-            addAll = { dslList, editedList -> dslList.addAll(editedList) }
-        )
+    suspend fun setFollowedTopicIds(followedTopicIds: Set<String>) {
+        try {
+            userPreferences.updateData {
+                with(it.toBuilder()) {
```

Comment:
Nit: There's a `copy` block here that does the same thing as `.toBuilder()` and `.build()` that I think looks slightly nicer?
```kotlin
it.copy {
    followedTopicIdsMap.clear()
    followedTopicIdsMap.putAll(followedTopicIds.associateWith { true })
}
```

Reply:
Probably me missing something obvious but when I try: 

```
it.copy { 
    followedTopicIdsMap.clear()
    followedTopicIdsMap.putAll(followedTopicIds.associateWith { true })
}
```

I get an `Collections$UnmodifiableMap` error from the following unit test `offlineFirstUserDataRepository_set_followed_topics_logic_delegates_to_nia_preferences`. 

I had to change it to: 

```
userPreferences.updateData {
                with(it.toBuilder()) {
                    followedTopicIdsMap.clear()
                    followedTopicIdsMap.putAll(followedTopicIds.associateWith { true })
                    build()
                }
            }
```

to get it to pass.

Reply:
Oh, maybe it needs to be this?

```
it.copy {
    this.followedTopicIds.clear()
    this.followedTopicIds.putAll(followedTopicIds.associateWith { true })
}
```

The `this` there is since the name in the `copy` DSL overlaps with the parameter